api/src/main/java/org/pmiops/workbench/google/CloudStorageClientImpl.java

```diff
@@ -177,4 +179,23 @@ public FileDetail blobToFileDetail(Blob blob, String bucketName, Set<String> wor
   public String getGoogleOAuthClientSecret() {
     return getCredentialsBucketString("google-oauth-client-secret.txt");
   }
+
+  @Override
+  public String notebookPathToFileDetail(String uri, Set<String> workspaceUsers) {
+    String bucketName = uri.replaceFirst("gs://", "");
+    String[] notebookPath = bucketName.split("/");
+    final String name =
+        Joiner.on('/').join(Arrays.copyOfRange(notebookPath, 1, notebookPath.length));
+    Blob blob = getBlob(notebookPath[0], name);
+    Map<String, String> fileMetadata = blob.getMetadata();
```

Comment:
This is a copy of the code in `blobToFileDetail()` - could you reuse that instead?

Please also rename this method.

Reply:
Done